src/client.js

```diff
@@ -788,8 +792,17 @@ export class StreamChat {
 			logger: this.logger,
 		});
 
+		let warmUp;
+		if (this.options.warmUp) {
+			warmUp = this.doAxiosRequest('options', this.baseURL + '/connect');
+		}
+
 		const handshake = await this.wsConnection.connect();
 		this.connectionID = this.wsConnection.connectionID;
+		if (this.options.warmUp) {
+			// wait for warm up connection to complete
+			await warmUp;
```

Comment:
I think a fire and forget is better for this

Reply:
done